test_suite/tests/derive.rs

```diff
@@ -609,6 +609,36 @@ fn doc_capture_works() {
     assert_type!(S, ty);
 }
 
+#[test]
+fn skip_type_params() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    #[scale_info(skip_type_params(T))]
+    struct Hey<T, N> {
+        ciao: Greet<T>,
+        ho: N,
+    }
+
+    #[derive(TypeInfo)]
+    #[scale_info(skip_type_params(T))]
+    struct Greet<T> {
+        marker: PhantomData<T>,
+    }
+
+    struct SomeType;
+
+    let ty = Type::builder()
+        .path(Path::new("Hey", "derive"))
+        .type_params(tuple_meta_type!(u16))
```

Comment:
it seems to me more correct to me to have the first type param marked as `skipped` than ignored, otherwise when looking at the metadata and the doc of the rust type we can't easily tell which one is skipped and which one is kept

Reply:
You mean like `.composite(Fields::named().field(|f| f.ty::<Greet<SomeType>>().name("ciao").type_name("Greet<T>").skipped_type_params::<T>())`?

Reply:
actually I'm not sure anymore, I don't understand enough the usecase for type_params.

But what I meant is that currently the `Hey<SomeType, u16>` type information about type params would be `u16`, whereas I would have expected `"skipped", u16`.

Reply:
This is a good point, we are losing some information by removing it entirely. I have addressed this by adding a `TypeParameter` struct with `Option<MetaType>` which would be `None` for skipped type params.